src/libsystemd/sd-journal/journal-file.c

```diff
@@ -3654,6 +3655,41 @@ int journal_file_open(
         return r;
 }
 
+int journal_file_parse_uid(const char *path, uid_t *ret_uid) {
```

Comment:
hmm, given the name of the function one might think this parses hte file. but it doens't, it parses information from the file *name*.

hence maybe:

journal_file_parse_uid_from_file_name()?